ledger/participant-state/kvutils/tools/src/main/scala/ledger/participant/state/kvutils/tools/integritycheck/IntegrityChecker.scala

```diff
@@ -214,6 +214,13 @@ class IntegrityChecker[LogResult](commitStrategySupport: CommitStrategySupport[L
               + s" submissionEnvelopeSize=${submissionInfo.submissionEnvelope.size()}"
               + s" writeSetSize=${expectedWriteSet.size}"
           )
+          expectedWriteSet.foreach {
```

Comment:
We'll need to verify write-set entries that are only supported by the integration kit (e.g., log fragments). Can you make this logic injectable/composable, please?

Reply:
Good point. I've moved it into `LogAppendingCommitStrategySupport` so we can implement it differently elsewhere.